examples/CMakeLists.txt

```diff
@@ -11,6 +11,12 @@
 #
 set(CMAKE_INSTALL_EXAMPLESDIR "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/examples")
 
+if(NOT BUILD_IDLC)
+  # We are cross-compiling the examples so we want to locate the host idlc
+  # The Generate.cmake file won't have been included yet
+  include("${CMAKE_SOURCE_DIR}/cmake/Modules/Generate.cmake")
```

Comment:
You can probably just `include("Generate.cmake")` because the directory is added to the module directories. If not, you may want to look into the difference between `CMAKE_SOURCE_DIR` and `PROJECT_SOURCE_DIR` for situations where Cyclone DDS is included as a submodule. I can imagine that's the case for ROS, but I might be wrong.